### PR TITLE
add emoji family for emoji clusters

### PR DESCRIPTION
**Pull request by dfrg**
Quick and dirty fallback for emoji. This can be done more efficiently (and should be handled in fontique directly) but this should get us functioning emoji mapping for capable renderers.

### CODE REVIEW

**Review by DJMcNab — DISMISSED** [review]
Is it meaningful to have a test for this?

cc @nicoburns for turning #55 and/or #54 also into integration tests

File: src/shape.rs [hunk 1]
```diff
@@ -211,8 +211,10 @@ impl<'a, 'b, B: Brush> partition::Selector for FontSelector<'a, 'b, B> {
     type SelectedFont = SelectedFont;
 
     fn select_font(&mut self, cluster: &mut CharCluster) -> Option<Self::SelectedFont> {
+        use fontique::QueryFamily;
         let style_index = cluster.user_data() as u16;
-        if style_index != self.style_index {
+        let is_emoji = cluster.info().is_emoji();
+        if style_index != self.style_index || is_emoji {
             self.style_index = style_index;
```
Comment (nicoburns) [inline src/shape.rs:218]: Do we also need a `self.is_emoji`? I can see this potentially failing to switch back to a regular font after an emoji...

File: src/shape.rs [hunk 2]
```diff
@@ -211,8 +211,10 @@ impl<'a, 'b, B: Brush> partition::Selector for FontSelector<'a, 'b, B> {
     type SelectedFont = SelectedFont;
 
     fn select_font(&mut self, cluster: &mut CharCluster) -> Option<Self::SelectedFont> {
+        use fontique::QueryFamily;
```
Comment (nicoburns) [inline src/shape.rs:214]: This should be at the top of the file with the other imports

**Comment by nicoburns** [comment]
Applying this on top of the tiny-skia example I get a blank space for the emoji (which is expected because the `ColorPainter` trait and associated function call aren't implemented):

Screenshot 2024-05-22 at 12 22 06

On top of the swash example I get "interesting" output:

Screenshot 2024-05-22 at 12 25 21

I'm assuming this is an issue with the example rather than this PR as it seems to selecting the correct font/glyph.

File: src/shape.rs (hunk 1 above)
Comment (dfrg) [reply src/shape.rs:218]: Good catch. Checking for `self.font_id.is_none()` should be sufficient. Ultimately I’d like to rework this logic.

Comment (dfrg) [reply src/shape.rs]: Added this additional check which will force update of the current font stack following an emoji cluster.

File: src/shape.rs (hunk 2 above)
Comment (dfrg) [reply src/shape.rs]: Done

**Comment by dfrg** [comment]
> Is it meaningful to have a test for this?

It is meaningful but the structure of this file makes unit testing difficult. I filed an issue for refactoring at #57

**Pull request merged**